src/node-binance-api.ts

```diff
@@ -638,6 +645,42 @@ export default class Binance {
         }
     }
 
+    unarmorKey(a:string):number[] {
+        // eslint-disable-next-line no-useless-escape
+        const m = /-----BEGIN [^-]+-----\n([A-Za-z0-9+\/=\s]+)\n-----END [^-]+-----|begin-base64[^\n]+\n([A-Za-z0-9+\/=\s]+)====/.exec(a);
+        if (m) {
+            if (m[1]) {
+                a = m[1];
+            } else if (m[2]) {
+                a = m[2];
+            } else {
+                throw new Error("RegExp out of sync");
+            }
+        }
+        return base64.decode(a);
+    }
+
+    generateSignature(query: string) {
+        const APISECRET = this.Options.APISECRET || this.APISECRET;
+        let signature = '';
+        if (APISECRET.includes ('PRIVATE KEY')) {
+            // if less than the below length, then it can't be RSA key
+            if (APISECRET.length < 500) {
```

Comment:
it's easier and safer to do the opposite, if the `apiSecret < 120` then we know it's ECDSA